scripts/component-base.js

```diff
@@ -250,14 +272,22 @@ export default class ComponentBase extends HTMLElement {
     let values = classToFlat(configByClasses);
 
     // get the external config
-
-    const configs = unflat(await externalConfig.getConfig(this.componentName, values.config));
-
-    if (!this.overrideExternalConfig) {
-      values = deepMerge({}, configs, values);
-    } else {
+    // TODO With the unFlatten approach of setting this.attributesValues there is an increased amount of processing
+    // each time a viewport changes when we need to flatten again the values
+    // better approach would be to generate this.attributesValues in the final state needed by each time of data:
+    // - class -  as arrays with unique values
+    // - data - as flatten values with camel case keys
+    // - attributes - as flatten values with hyphen separated keys.
+    // for anything else set them flatten as they come from from external config
```

Comment:
should we create a ticket for this?

Reply:
Yes we should, I will add it as a topic for the refinement meeting